### PR TITLE
Change volume -> pool

### PR DESCRIPTION
**Pull request by aravindavk**

### CODE REVIEW

File: docs/pools.adoc [hunk 1]
```diff
@@ -10,43 +10,27 @@ Enable and start the `kadalu-mgr`
...
-[root@server1 ~]# kadalu pool list
-                                  ID  Name
-4e67989e-0c24-44ac-b0a7-81aa5cb4ec6b  DEV
-----
-
-The pool data and the configurations are stored in `/var/lib/kadalu`. Since the storage manager is running in a single node, by default it is not highly available. Use `kadalu-lite` Volume for high availability, or backup the Pool data in regular intervals. Refer "High availability" chapter (Coming soon) for more details.
+The cluster configurations are stored in `/var/lib/kadalu`. Since the storage manager is running in a single node, by default it is not highly available. Use `kadalu-lite` Volume for high availability, or backup the Pool data in regular intervals. Refer "High availability" chapter (Coming soon) for more details.
 
-Since Kadalu Storage manager supports multiple Pools, the manager node is not part of any Pool by default. Optionally invite the Storage manager node to join the Pool, which is similar to inviting other Storage nodes.
+No nodes are added to the cluster by default. Invite the node to become part of the Cluster by running the following command.
```
Comment (vatsa287) [inline docs/pools.adoc]: Does manager node be part of of any pool by default.?
If manager node is invited be a part of particular pool, can it also manage other pools?

File: docs/volumes.adoc [hunk 2]
```diff
@@ -1,155 +1,156 @@
...
+[root@server1 ~]# kadalu pool create pool1 server1.example.com:/exports/pool1/s1
+Pool pool1 created successfully
 ID: 02b6a3a4-e704-47ee-bb05-e541a561a921
 ----
 
 Where
 
-- `DEV` is Pool name
-- `vol1` is the Volume name
+- `vol1` is the Pool name
```
Comment (vatsa287) [inline docs/volumes.adoc]: `pool1`?

File: docs/volumes.adoc [hunk 3]
```diff
@@ -1,155 +1,156 @@
...
 - `server1.example.com` is the node from which the Storage unit will be exported.
 - `/exports/vol/s1` is the mounted path of Storage Unit.
 
-Automatically create the Pool and add nodes to Pool that are specified in the Volume create request by adding `--auto-create-pool` and `--auto-add-nodes`.
+Automatically add nodes to the Cluster that are specified in the Pool create request by adding `--auto-add-nodes`.
```
Comment (vatsa287) [inline docs/volumes.adoc]: Here cluster refers to group of pools, or the pool itself?

File: mgr/src/cmds/pools.cr [hunk 4]
```diff
@@ -1,79 +1,345 @@
...
+    if pools.size == 0
+      puts "No pools available in the Cluster.\nRun `kadalu pool create \"<pool>\" ...` to create a pool."
+    end
+
+    if args.pool_args.detail
+      pools.each do |pool|
+        pool_detail(pool, args)
+      end
+
+      next
```
Comment (vatsa287) [inline mgr/src/cmds/pools.cr:191]: Usage of next here?

File: mgr/src/cmds/pools.cr [hunk 5]
```diff
@@ -1,79 +1,345 @@
...
+
+command "pool.create", "Kadalu Storage Pool Create" do |parser, args|
+  parser.banner = "Usage: kadalu pool create NAME TYPE STORAGE_UNITS [arguments]"
+  parser.on("--no-start", "Don't start the pool upon creation") do
+    args.pool_args.no_start = true
+  end
+  parser.on("--pool-id=ID", "Set pool-id to import a pool") do |pool_id|
+    args.pool_args.pool_id = pool_id
+  end
+  parser.on("--distribute", "Create distributed Pool") do
```
Comment (vatsa287) [inline mgr/src/cmds/pools.cr]: Documentation on usage of '--distribute`

File: mgr/src/server/datastore/nodes.cr [hunk 6]
```diff
@@ -6,90 +6,67 @@ require "db"
...
-
-  def list_nodes(pool_name)
-    grouped_nodes(connection.query_all(nodes_query + " WHERE pools.name = ?" + nodes_query_order_by, pool_name, as: NodeView))
-  end
+  #   query = nodes_query + " WHERE pools.id IN (#{(pool_ids.map { |_| "?" }).join(",")}) " + nodes_query_order_by
+  #   connection.query_all(query, args: pool_ids, as: MoanaTypes::Node)
+  # end
 
-  def get_nodes(pool_name, node_names)
+  def get_nodes(node_names)
```
Comment (vatsa287) [inline mgr/src/server/datastore/nodes.cr:36]: Now a node can be part of different pools?

File: mgr/src/server/datastore/roles.cr [hunk 7]
```diff
@@ -4,19 +4,7 @@ module Datastore
...
-
-  def viewable_volume_ids(user_id, pool_id)
-    query = "SELECT DISTINCT volume_id FROM roles WHERE user_id = ? AND (pool_id = ? OR pool_id = ?)"
-    connection.query_all(query, user_id, pool_id, "all", as: String)
-  end
-
-  def role?(user_id, pool_name, volume_name, roles)
+  def role?(user_id, pool_name, roles)
     # If pool_name is "all" then do not join the pools table
     # Similarly do not join volumes table if the volume name is "all"
```
Comment (vatsa287) [inline mgr/src/server/datastore/roles.cr]: Think this can be removed. L#9

**Review by vatsa287 — CHANGES_REQUESTED** [review]

**Comment by vatsa287** [comment]
Going by this PR, now we would have to implement `subdir` mounts to carve out volumes from this pool.
And Rebalance feature would only be used when pools are made of type `--distribute` or would cease to exist?

File: docs/pools.adoc (hunk 1 above)
Comment (aravindavk) [reply docs/pools.adoc]: nodes are not limited to a pool now. A node can be part of one or more storage pools

File: docs/volumes.adoc (hunk 2 above)
Comment (aravindavk) [reply docs/volumes.adoc]: Ack

File: docs/volumes.adoc (hunk 3 above)
Comment (aravindavk) [reply docs/volumes.adoc]: Cluster is group of nodes that can communicate each other. Pools are created using the nodes of a Cluster.

File: mgr/src/cmds/pools.cr (hunk 5 above)
Comment (aravindavk) [reply mgr/src/cmds/pools.cr]: Not yet implemented in API side. I will remove this from here and add to separate PR. The idea of this flag is to make the user aware that the pool that is going to be created is distributed. Current behaviour is to automatically decide if more storage units are provided than the Replica/Disperse count.

File: mgr/src/server/datastore/nodes.cr (hunk 6 above)
Comment (aravindavk) [reply mgr/src/server/datastore/nodes.cr:36]: Yes. If a node consists of two storage units, one can be used with one pool and the other one can be used by the second pool.

File: mgr/src/server/datastore/roles.cr (hunk 7 above)
Comment (aravindavk) [reply mgr/src/server/datastore/roles.cr]: Ack

**Comment by aravindavk** [comment]
> Going by this PR, now we would have to implement `subdir` mounts to carve out volumes from this pool. And Rebalance feature would only be used when pools are made of type `--distribute` or would cease to exist?

Yes. Rebalance is applicable only for distributed Pools or when an existing Pool is expanded by running pool expand command. `--distribute` flag is to avoid confusion to users and prevent creating distributed pools unknowingly.

File: mgr/src/cmds/pools.cr (hunk 4 above)
Comment (aravindavk) [reply mgr/src/cmds/pools.cr:191]: same as `return` in function. Use `next` in blocks. This file is renamed from volumes.cr to pools.cr and then Volume changed to Pool. Diff is showing it as full new implementation :(

**Comment by aravindavk** [comment]
Changes updated.

**Review by vatsa287 — APPROVED** [review]
LGTM

**Pull request merged**